File: app/src/main/java/com/vsa/filmoteca/presentation/usecase/GetMoviesListUseCase.java
package com.vsa.filmoteca.presentation.usecase;

import com.vsa.filmoteca.data.domain.Movie;
import com.vsa.filmoteca.data.repository.MoviesRepository;

import java.util.List;

import rx.Observable;

/**
 * Created by albertovecinasanchez on 18/7/16.
 */

public class GetMoviesListUseCase {

    private MoviesRepository mMoviesRepository;

    public GetMoviesListUseCase(MoviesRepository moviesRepository) {
        mMoviesRepository = moviesRepository;
    }

    public Observable<List<Movie>> moviesList() {
        return mMoviesRepository.moviesList();
    }

}

File: app/src/main/java/com/vsa/filmoteca/domain/usecase/GetMoviesListUseCase.kt
package com.vsa.filmoteca.domain.usecase

import com.vsa.filmoteca.data.source.repository.MoviesDataRepository
import com.vsa.filmoteca.domain.model.Movie
import com.vsa.filmoteca.network.executor.AsyncExecutor
import javax.inject.Inject

/**
 * Created by albertovecinasanchez on 18/7/16.
 */

class GetMoviesListUseCase @Inject constructor(
    private val repository: MoviesDataRepository,
    private val asyncExecutor: AsyncExecutor
) {

    fun moviesList(callback: (Result<List<Movie>>) -> Unit) {
        asyncExecutor.execute({
            repository.moviesList()
        }, callback)
    }

}
